src/MySql/custom/New-AzMySqlFlexibleServer.ps1

```diff
@@ -147,60 +183,418 @@ function New-AzMySqlFlexibleServer {
+            }
+            $RequiredKeys = 'SubscriptionId', 'ResourceGroupName', 'Name', 'Location'
+            ForEach($Key in $RequiredKeys){ $NetworkParameters[$Key] = $PSBoundParameters[$Key] }
+
+            If(!$NetworkParameters.ContainsKey('PublicAccess')){
+                $VnetSubnetParameters = CreateNetworkResource $NetworkParameters
+                $SubnetId = GetSubnetId $VnetSubnetParameters.ResourceGroupName $VnetSubnetParameters.VnetName $VnetSubnetParameters.SubnetName
+                $PSBoundParameters.DelegatedSubnetArgumentSubnetArmResourceId = $SubnetId
+            }
+            If ([string]::IsNullOrEmpty($PSBoundParameters.DelegatedSubnetArgumentSubnetArmResourceId)) {
+                $null = $PSBoundParameters.Remove('DelegatedSubnetArgumentSubnetArmResourceId')
+            }
+
+            $Msg = 'Creating MySQL server {0} in group {1}...' -f $PSBoundParameters.Name, $PSBoundParameters.resourceGroupName
+            Write-Host $Msg
+            $Msg = 'Your server {0} is using sku {1} (Paid Tier). Please refer to https://aka.ms/mysql-pricing for pricing details' -f $PSBoundParameters.Name, $PSBoundParameters.SkuName
+            Write-Host $Msg
+            $Server = Az.MySql.internal\New-AzMySqlFlexibleServer @PSBoundParameters
+
+            # # Create Database
+            $DatabaseParameter = [Microsoft.Azure.PowerShell.Cmdlets.MySql.Models.Api20171201.Database]::new()
+            $DatabaseParameter.Charset = "utf8"
+            $DatabaseParameter.Collation = "utf8_general_ci"
+            $Msg = 'Creating database {0}...' -f $DEFAULT_DB_NAME
+            Write-Host $Msg
+            $null = New-AzMySqlFlexibleServerDatabase -Name $DEFAULT_DB_NAME -ResourceGroupName $PSBoundParameters.ResourceGroupName -ServerName $PSBoundParameters.Name -Parameter $DatabaseParameter
+            
+            # Create Firewallrules
+            $FirewallRuleName = CreateFirewallRule $NetworkParameters
+
+            $Msg = "
+`"databaseName`": `"$DEFAULT_DB_NAME`",
+`"id`": `"/subscriptions/$($PSBoundParameters.SubscriptionId)/resourceGroups/$($PSBoundParameters.ResourceGroupName)/providers/Microsoft.DBForMySql/flexibleServers/$($PSBoundParameters.Name)`",
+`"location`": `"$($PSBoundParameters.Location)`",
+`"password`": `"$($PSBoundParameters.AdministratorLoginPassword)`",
+`"resourceGroup`": `"$($PSBoundParameters.ResourceGroupName)`",
+`"skuname`": `"$($PSBoundParameters.SkuName)`",
+`"username`": `"$($PSBoundParameters.AdministratorLogin)`",
+`"version`": `"$($PSBoundParameters.Version)`"
+"
+
+            If ($PSBoundParameters.ContainsKey('DelegatedSubnetArgumentSubnetArmResourceId')) {
+                $Msg += "`"subnetId`": `"$($PSBoundParameters.DelegatedSubnetArgumentSubnetArmResourceId)`""
+            }
+            ElseIf (![string]::IsNullOrEmpty($FirewallRuleName)) {
+                $Msg += "`"firewallName`": `"$FirewallRuleName`""
+            }
+            Write-Host $Msg
+
+            return $Server
```

Comment:
Add all the field you want to print in the directlve.

Reply:
New-AzMySqlFkexibleServer: I want to show  the password, firewall rule name, database name that are not available in the server object. 
By this: "Add all the field you want to print in the directlve to type server", Do you mean that I can add this to the returned object? Or is there any way I can add that??

Reply:
You can create a PSCustomObject which includes all the properties you want and return it.

Reply:
As we disscused, you can create a new class implements IServerAutoGenerated using c#, add the customerized property you want and use it in New-AzMySqlFlexibleServer. Please refer to the custom folder of CloudService

Reply:
@DaeunYim  check with the codes, find that MySqlIdentity is not the base class of IServerAutoGenerated. It can be converted from some specified types.

```csharp
            if (null == sourceValue)
            {
                return true;
            }
            global::System.Type type = sourceValue.GetType();
            // we allow string conversion too.
            if (type == typeof(global::System.String))
            {
                return true;
            }
            if (typeof(global::System.Management.Automation.PSObject).IsAssignableFrom(type))
            {
                // we say yest to PSObjects
                return true;
            }
            if (typeof(global::System.Collections.IDictionary).IsAssignableFrom(type))
            {
                // we say yest to Hashtables/dictionaries
                return true;
            }
```
You could try to create a psobject rather than pscustomobject